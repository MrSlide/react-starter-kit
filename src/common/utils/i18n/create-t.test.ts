import Polyglot from 'node-polyglot'
import createT from './create-t'

describe('i18n createT()', function () {
  const langCode = 'en-GB'
  const phrases = {
    test: {
      phrase: 'Hello, world!',
      placeholder: 'Hello, %{name}!',
      pluralization: '%{smart_count} test |||| %{smart_count} tests'
    }
  }
  const t = createT(langCode, phrases)

  test('returns a t function', function () {
    expect(t.constructor).toBe(Polyglot.prototype.t.constructor)
  })

  describe('returned t function', function () {
    test('provides localized strings', function () {
      const output = t('test.phrase')

      expect(output).toBe('Hello, world!')
    })

    test('provides localized strings with replaced placeholders', function () {
      const output = t('test.placeholder', {
        name: 'Test'
      })

      expect(output).toBe('Hello, Test!')
    })

    test('provides localized strings with pluralization', function () {
      const outputSingular = t('test.pluralization', {
        smart_count: 1
      })
      const outputPlural = t('test.pluralization', {
        smart_count: 2
      })

      expect(outputSingular).toBe('1 test')
      expect(outputPlural).toBe('2 tests')
    })

    test('throws an error on missing phrases', function () {
      expect(function () {
        t('test.missing')
      }).toThrowErrorMatchingSnapshot()
    })
  })
})
