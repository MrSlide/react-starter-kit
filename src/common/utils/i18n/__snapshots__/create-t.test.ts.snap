// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18n createT() returned t function throws an error on missing phrases 1`] = `"The translation key 'test.missing' is missing for 'en-GB'"`;
